[SAM] Sen Overhaul

Rjb99 opened the pull request:
This PR contains the update to Sen.tsx.

Instead of just saying how you used Sen, the module now points out in visual format when, how and what caused you to use Sen in a non-standard use.
 
example output:
![image](https://user-images.githubusercontent.com/37195907/81457052-bbb14180-9162-11ea-827d-49688e2abe51.png)

AzariahAeroborn commented on src/parser/jobs/sam/modules/Sen.tsx:
```diff
@@ -1,98 +1,307 @@
+
 export default class Sen extends Module {
-	static handle = 'sen'
+	static handle = 'Non-Standard Sen Windows'
```
Handle should be a single camelCase “word”, in case other modules need to depend on this one. 

To change what displays in the sidebar, use the title static property and a trans tag or the t() macro -- e.g. 	
```suggestion
	static handle = 'sen'
	static title = t('sam.sen.title')`Non-Standard Sen Windows`
```

To use the t() macro, you will need to also `import {t} from '@lingui/macro'`

AzariahAeroborn commented on src/parser/jobs/sam/modules/Sen.tsx:
```diff
@@ -1,98 +1,307 @@
+
+		// step 2: FILTER EVERYTHING
+
+		if (THINGS_WE_WANT_IN_THE_TABLE.includes(action) || SEN_ACTIONS.includes(action) || SEN_REMOVERS.includes(action) ) {
```
Redundant check, since we’re applying the first filter in the event hook and the other two filters will never get hit.

AzariahAeroborn commented on src/parser/jobs/sam/modules/Sen.tsx:
```diff
@@ -1,98 +1,307 @@
+		const lastSenState = this.lastSenState
+
+		if (lastSenState != null && lastSenState.end == null) {
+			this.wasted = this.wasted + (lastSenState.overwriteSetsus + lastSenState.overwriteGetsus + lastSenState.overwriteKas) + (lastSenState.currentSetsu + lastSenState.currentGetsu + lastSenState.currentKa)
```
Even if there were no Sen lost due to the death, this is still marking the Sen window as non-standard due to a death and adding it to the output table (see http://localhost:3000/analyse/dvhD12atwZFLz8AN/3/11/).  Should those cases be excluded from the output table?

AzariahAeroborn commented on src/parser/jobs/sam/modules/Sen.tsx:
```diff
@@ -110,4 +319,78 @@ export default class Sen extends Module {
+				<Message warning icon>
+					<Icon name="warning sign"/>
+					<Message.Content>
+						<Trans id = "sam.sen.rotation-table.disclaimer">This module labels a "Standard Sen Window" to be a window with no Sen overwrites that ends on a Iaijutsu. Please consult The Balance Discord's guides for more details on looping Samurai gameplay.
```
Nit: "ends on **an** Iaijutsu

Also, is there a specific Balance guide we can link to in the recommendation?

Rjb99 replied:
Fixed.

Rjb99 commented:
I've updated the PR.

Fixed Aza's requests, along with some balance sam feedback thanks to nono.

Overwote sens is now spilt into 2, single overwrite and double overwrite

Haga is also spilt into 2, single sen converts aka Standards, and multi sen converts aka Non-Standards

Rjb99 commented on src/parser/jobs/sam/modules/Sen.tsx:
```diff
@@ -1,105 +1,339 @@
+
+	kenkiGained: number = 0 // Kenki # * 10
+
+	_senCode: {priority: number, message: TODO} = SEN_HANDLING.NONE
```
Going to assume no, because it crashes if I try to assign the error code string to the JSX.Element due to the string not being assignable.

Rjb99 replied:
Nevermind, Done. as Aza told me in discord, changing the "No Errors" to <> Error Code </> let me change it to JSX.Element.
